magma/view.py

```diff
@@ -43,13 +45,16 @@ def __getattr__(self, key):
             return object.__getattribute__(self, key)
         return PortView[type(attr)](attr, self.parent)
 
-    def get_hierarchical_coreir_select(self):
+    def path(self) -> Tuple[str]:
+        path = [self.port.name.name]
         curr = self.parent
-        hierarchical_path = curr.inst.name + ";"
-        while isinstance(curr.parent, InstView):
-            hierarchical_path = curr.parent.inst.name + ";" + hierarchical_path
+        while isinstance(curr, InstView):
+            path.insert(0, curr.inst.name)
             curr = curr.parent
-        return hierarchical_path
+        return tuple(path)
```

Comment:
Could use tuple addition instead of insert then tuple conversion to avoid the intermediate list

e.g.
```
path = (self.port.name, )
...
    path = (curr.inst.name, ) + path
```